test/index.spec.js

```diff
@@ -1941,4 +1888,450 @@ describe('discipl-law-reg', () => {
+      }
+
+      expect(Object.keys(completeFacts)).to.not.include('[bedrag]')
+      expect(acts).to.not.deep.include({ 'act': '<<bedrag vaststellen>>', 'actor': 'ouder' })
+      expect(errorMessage).to.equal('Action <<aanvraag kinderbijslag toekennen>> is not allowed')
+    })
+  })
+
+  describe.only('PROJECTION expression', async function () {
+    it('should get the projected property', async () => {
+      const model = {
+        'acts': [
+          {
+            'act': '<<subsidie aanvragen>>',
+            'actor': '[burger]',
+            'recipient': '[ambtenaar]',
+            'object': '[verzoek]',
+            'preconditions': {
+              'expression': 'AND',
+              'operands': [
+                '[bedrag]'
+              ]
+            },
+            'create': [
+              '[aanvraag]'
+            ]
+          },
+          {
+            'act': '<<subsidie aanvraag toekennen>>',
+            'actor': '[ambtenaar]',
+            'object': '[aanvraag]',
+            'preconditions': {
+              'expression': 'EQUAL',
+              'operands': [
+                {
+                  'expression': 'PROJECTION',
+                  'context': [
+                    '[aanvraag]'
+                  ],
+                  'fact': '[bedrag]'
+                },
+                {
+                  'expression': 'LITERAL',
+                  'operand': 500
+                }
+              ]
+            },
+            'recipient': '[burger]'
+          }
+        ],
+        'facts': [
+          {
+            'fact': '[aanvraag]',
+            'function': {
+              'expression': 'CREATE',
+              'operands': [
+                '[bedrag]'
+              ]
+            }
+          }
+        ],
+        'duties': []
+      }
+
+      const completeFacts = { '[burger]': true, '[ambtenaar]': true, '[verzoek]': true, '[bedrag]': 500 }
+      const util = new Util(lawReg)
+
+      const { ssids, modelLink } = await util.setupModel(model, ['burger', 'ambtenaar'], { 'burger': '[burger]', 'ambtenaar': '[ambtenaar]' })
+      const acts = [
+        {
+          'act': '<<subsidie aanvragen>>',
+          'actor': 'burger'
+        },
+        {
+          'act': '<<subsidie aanvraag toekennen>>',
+          'actor': 'ambtenaar'
+        }
+      ]
+
+      let errorMessage = ''
+      try {
+        await util.scenarioTest(ssids, modelLink, acts, completeFacts)
+      } catch (e) {
+        errorMessage = e.message
+      }
+
+      expect(completeFacts).to.deep.include({ '[bedrag]': 500 })
+      expect(errorMessage).to.equal('')
+    })
+
+    it('should project a series of CREATE facts', async () => {
+      const model = {
+        'acts': [
+          {
+            'act': '<<persoonlijk gegevens invullen>>',
+            'actor': '[burger]',
+            'recipient': '[ambtenaar]',
+            'object': '[verzoek]',
+            'preconditions': {
+              'expression': 'AND',
```

Comment:
Any particular reason this has the "AND" operator wrapping the operand?

Reply:
No :) Usage of `AND` is indeed redundant in this case